src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -178,12 +178,17 @@ private protected virtual IIntellisense CreateIntellisense()
             return IntellisenseProvider.GetIntellisense(Config);
         }
 
+        public IIntellisenseResult Suggest(string expression, RecordType parameterType, int cursorPosition)
+        {
+            return Suggest(expression, parameterType, cursorPosition, null);
+        }
+
         /// <summary>
-        /// Get intellisense from the formula.
+        /// Get intellisense from the formula, with parser options.
         /// </summary>
-        public IIntellisenseResult Suggest(string expression, RecordType parameterType, int cursorPosition)
+        public IIntellisenseResult Suggest(string expression, RecordType parameterType, int cursorPosition, ParserOptions options)
```

Comment:
>IIntellisenseResult Suggest(string expression, RecordType parameterType, int cursorPosition, ParserOptions options) [](http://example.com/codeflow?start=15&length=115)

Can we just have an overload that takes CheckResult?

Something like:
Suggest(string expression, CheckResult check, int cursorPosition);

That may be the most flexible - since there are other options for suggest (and future options on the way) - see the overloads of Check.

Reply:
The caller will have the obligation to call Check() before but that seems a reasonable approach and allows access to the potential errors that could occur

Reply:
exactly. By allowing the caller to call Check(), that means they can access all the various flags/overloads that check providers (without having to duplicate here)